torch/distributed/device_mesh.py

```diff
@@ -376,10 +389,16 @@ def __getitem__(self, mesh_dim_name: str) -> "DeviceMesh":
                         f"Invalid mesh_dim_name {mesh_dim_name} specified."
                     )
 
-            mesh_dim = _mesh_resources.get_mesh_dim_by_name(self, mesh_dim_name)
-            submesh = _mesh_resources.create_child_mesh(self, mesh_dim, mesh_dim_name)
-
-            return submesh
+            # Directly return the child mesh if it is already created.
+            submesh = _mesh_resources.get_child_mesh(self, mesh_dim_name)
```

Comment:
we don't need these checks if we just put the cache logic into `child_mesh` directly